components/openzwave-adapter/src/lib.rs

```diff
@@ -346,11 +372,17 @@ impl OpenzwaveAdapter {
                         // https://github.com/fxbox/taxonomy/issues/97
                         // When it's done we can move the properties change from above to here.
                     }
-                    ZWaveNotification::NodeRemoved(_node)           => {}
+                    ZWaveNotification::NodeRemoved(node)           => {
+                        if let Some(service_id) = node_map.find_taxo_id_from_ozw(&node) {
```

Comment:
here you can use `remove_by_ozw` if it returns the removed item as advised above (is there a reason you don't remove the item from node_map here BTW?)